### DISCUSSION TITLE
Import foxglove 3d panel into an external application

### DISCUSSION THREAD

**Discussion by AriannaC10** [foxglove/community#736] [General]
Hi everyone, thank you for your codes and effort! Great idea to create some extensions to embed customize panels into foxglove studio.

I was wondering if there is any chance it can developed something to import one or more of the foxglove panels into external applications. 

This can be useful for the project I am working on with my team. The panel that we are interested in importing in our application is the 3D panel, the application is written in React and it's communicating with ROS2. 

Thank you in advance for your time, answers and suggestions!

**Comment by defunctzombie** [comment]
Right now there is no way to take individual panels and bring them into other applications and we have no plans for that workflow on our roadmap; we focus on developing Studio as a complete app itself. Since the code for these panels is open source you could probably figure out some ways to extract or re-use our app components.
